File: examples/Python/mdclient2.py
"""
Majordomo Protocol client example. Uses the mdcli API to hide all MDP aspects

Author : Min RK <benjaminrk@gmail.com>

"""

import sys
from mdcliapi2 import MajorDomoClient

def main():
    verbose = '-v' in sys.argv
    client = MajorDomoClient("tcp://localhost:5555", verbose)
    requests = 100000
    for i in xrange(requests):
        request = "Hello world"
        try:
            client.send("echo", request)
        except KeyboardInterrupt:
            print "send interrupted, aborting"
            return
    
    count = 0
    while count < requests:
        try:
            reply = client.recv()
        except KeyboardInterrupt:
            break
        else:
            # also break on failure to reply:
            if reply is None:
                break
        count += 1
    print "%i requests/replies processed" % count
    
if __name__ == '__main__':
    main()

File: examples/Ruby/mdclient2.rb
# Majordomo Protocol client example. Uses the mdcliapi2 API to hide all MDP aspects
#
# Author : Tom van Leeuwen <tom@vleeuwen.eu>
# Based on Python example by Min RK

require './mdcliapi2.rb'

client = MajorDomoClient.new('tcp://localhost:5555')
requests = 100000
requests.times do |i|
  request = 'Hello world'
  begin
    client.send('echo', request)
  end
end

count = 0
while count < requests do
  begin
    reply = client.recv
  end
  count += 1
end

puts "#{count} requests/replies processed"
